cli/cli.go

```diff
@@ -49,11 +50,15 @@ var initialize = &cobra.Command{
 		// TODO: Synchronize and do not write before peer signal.Send has finished/peer is attached
 		// Peer needs to signal readiness
 
-		if err := s.Attach(args[0], args[1:]...); err != nil {
+		if err := s.Run(args[0], args[1:]...); err != nil {
 			panic(err)
 		}
 
-		if err := s.Loop(); err != nil {
+		if err := s.Run(args[0], args[1:]...); err != nil {
+			panic(err)
+		}
+
+		if err := s.Err(); err != nil {
```

Comment:
Because `s.err` is set in a different go-routine (`go s.Loop()`) this should probably be synchronized.